### PR TITLE
feat: add support for request timeout settings in the HTTP request node.

### LINKED ISSUE #3806: The HTTP request workflow node supports setting timeout parameters.
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

I am using an HTTP request node to call a service which takes a long time to respond. Currently, the timeout is hardcoded to 60 seconds in the code, which causes the service call to fail. I hope to make it configurable.

### 2. Describe the feature you'd like to see

I hope to make it configurable.

### 3. How will this feature improve your workflow or experience?

Successfully called the external service I provided

### 4. Additional context or comments

_No response_

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by majian159**
# Description

Backend
1. Add Timeout type and timeout field to HttpRequestNodeData
2. Modify the HTTP_REQUEST_DEFAULT_TIMEOUT setting to have a connection timeout of 10 seconds, read timeout of 60 seconds, and write timeout of 20 seconds
3. Apply the timeout configuration on httpx

Frontend
1. Add readOnly, min, max attributes to the Input component
2. Add Timeout settings to the HTTP Request Node
3. Add i18n information

Fixes #3806 #3808

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I am debugging locally to ensure that the timeout parameter is passed into httpx and that the request is successful.

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

File: web/i18n/en-US/workflow.ts [hunk 1]
```diff
@@ -249,6 +249,15 @@ const translation = {
         'header': 'Header',
       },
       insertVarPlaceholder: 'type \'/\' to insert variable',
+      timeout: {
+        title: 'Timeout',
+        connectLabel: 'Connect Timeout',
```
Comment (crazywoola) [inline web/i18n/en-US/workflow.ts]: I suggest to Connection Timeout here.

**Comment by crazywoola** [comment]
Hello thanks for your contribution, please add my wechat: crazyphage. I will invite you to our developer group chat.

File: web/app/components/workflow/nodes/http/use-config.ts [hunk 2]
```diff
@@ -148,6 +155,8 @@ const useConfig = (id: string, payload: HttpNodeType) => {
     showAuthorization,
     hideAuthorization,
     setAuthorization,
+    // timeout
```
Comment (crazywoola) [inline web/app/components/workflow/nodes/http/use-config.ts]: Please remove the unused code.

**Review by crazywoola — COMMENTED** [review]
Please fix the errors in CI as well.

**Comment by Yeuoly** [comment]
I suggest a max timeout to avoid network blocking issues, otherwise it brings a memory leak problem.

**Comment by majian159** [comment]
> I suggest a max timeout to avoid network blocking issues, otherwise it brings a memory leak problem.

The timeout period can vary depending on different business scenario requirements. Should we impose this limit on the frontend or the backend? How does 10 minutes sound?

**Comment by Yeuoly** [comment]
> > I suggest a max timeout to avoid network blocking issues, otherwise it brings a memory leak problem.
> 
> The timeout period can vary depending on different business scenario requirements. Should we impose this limit on the frontend or the backend? How does 10 minutes sound?

I prefer using environment variables to specify default max timeout of it, how do you think of this? we can use 10 minutes by default.

**Comment by majian159** [comment]
> > > 我建议最大超时以避免网络阻塞问题，否则会带来内存泄漏问题。
> > 
> > 
> > 根据不同的业务场景需求，超时时间可能会有所不同。我们应该对前端还是后端施加这个限制？ 10 分钟听起来怎么样？
> 
> 我更喜欢使用环境变量来指定它的默认最大超时，您对此有何看法？我们可以默认使用 10 分钟。

Should we use environment variables to set the timeout, taking into account the maximum value allowed by the frontend component? Or should we return a 400 error at the API level?

**Comment by majian159** [comment]
> > > I suggest a max timeout to avoid network blocking issues, otherwise it brings a memory leak problem.
> > 
> > 
> > The timeout period can vary depending on different business scenario requirements. Should we impose this limit on the frontend or the backend? How does 10 minutes sound?
> 
> I prefer using environment variables to specify default max timeout of it, how do you think of this? we can use 10 minutes by default.

If we use environment variables to set the timeout, we need to consider the maximum value allowed by the frontend components. Or should we return a 400 error at the API layer instead?

**Comment by majian159** [comment]
@Yeuoly 
Add environment variables HTTP_REQUEST_MAX_CONNECT_TIMEOUT, HTTP_REQUEST_MAX_READ_TIMEOUT, HTTP_REQUEST_MAX_WRITE_TIMEOUT to limit the maximum timeout duration of HTTP requests. The default values are respectively 5 minutes, 10 minutes, and 10 minutes.

**Review by Yeuoly — APPROVED** [review]
LGTM

**Pull request merged**